src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts

```diff
@@ -0,0 +1,199 @@
+        this.loading = false;
+        this.configForm.setValue(configs);
+      },
+      (_error) => {
+        this.error = true;
+      }
+    );
+  }
+
+  createConfigForm() {
+    const controlsConfig = {};
+    _.forEach(Object.values(this.options), (option) => {
+      controlsConfig[option.name] = [option.default_value, this.getValidators(option)];
+    });
+    this.configForm = this.formBuilder.group(controlsConfig);
+  }
+
+  createPreviewForm() {
+    const controls = {
+      report: JSON.stringify(this.report, null, 2),
+      reportId: this.reportId,
+      licenseAgrmt: [this.licenseAgrmt, Validators.requiredTrue]
+    };
+    this.previewForm = this.formBuilder.group(controls);
+  }
+
+  download(report: object, fileName: string) {
+    this.textToDownloadService.download(JSON.stringify(report, null, 2), fileName);
+  }
+
+  getValidators(option: any): ValidatorFn[] {
+    const result = [];
+    switch (option.type) {
+      case 'int':
+        result.push(CdValidators.number());
+        result.push(Validators.required);
+        if (_.isNumber(option.min)) {
+          result.push(Validators.min(option.min));
+        }
+        if (_.isNumber(option.max)) {
+          result.push(Validators.max(option.max));
+        }
+        break;
+      case 'str':
+        if (_.isNumber(option.min)) {
+          result.push(Validators.minLength(option.min));
+        }
+        if (_.isNumber(option.max)) {
+          result.push(Validators.maxLength(option.max));
+        }
+        break;
+    }
+    return result;
+  }
+
+  disableModule(message: string = null, followUpFunc: Function = null) {
+    this.telemetryService.enable(false).subscribe(() => {
+      if (message) {
+        this.notificationService.show(NotificationType.success, this.i18n(message));
+      }
+      if (followUpFunc) {
+        followUpFunc();
+      } else {
+        this.router.navigate(['']);
+      }
+    });
+  }
+
+  getReport() {
+    this.loading = true;
+    this.telemetryService.getReport().subscribe(
+      (resp: object) => {
+        this.loading = false;
+        this.report = resp;
+        this.reportId = resp['report']['report_id'];
+        this.createPreviewForm();
+        this.step++;
+      },
+      (_error) => {
+        this.error = true;
+      }
+    );
+  }
+
+  updateConfig() {
+    const config = {};
+    _.forEach(Object.values(this.options), (option) => {
+      const control = this.configForm.get(option.name);
+      // Append the option only if the value has been modified.
+      if (control.dirty && control.valid) {
+        config[option.name] = control.value;
+      }
+    });
+    this.mgrModuleService.updateConfig('telemetry', config).subscribe(
+      () => {
+        this.disableModule(
+          'Your settings have been applied successfully. ' +
+            'Due to security reasons the Telemetry module is now disabled until you ' +
+            'completed the next step and accepted the license.',
```

Comment:
I would leave it as it is (at least for the moment) because the message does only appear if the user changed at least one of the configuration parameters. If others will complain about it too, we can think about putting it somewhere else at a later point.